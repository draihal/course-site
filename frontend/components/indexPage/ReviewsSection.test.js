import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import TestReviewsSection from './ReviewsSection'

const reviews = [
  {
    "id": 2,
    "url": "/api/v1/pages/reviews/2/",
    "user_id": 2,
    "student_first_name": "Student2",
    "student_last_name": "Test2",
    "student_image": null,
    "text": "Тестовый отзыв 2"
  },
  {
    "id": 1,
    "url": "/api/v1/pages/reviews/1/",
    "user_id": 1,
    "student_first_name": "Student1",
    "student_last_name": "Test1",
    "student_image": null,
    "text": "Тестовый отзыв 1"
  }
];


describe('ReviewsSection', () => {
  it('render correctly ReviewsSection component', () => {
    const tree = renderer.create(<TestReviewsSection reviews={reviews} />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('render ReviewsSection correctly with given props', () => {
    const { queryByText } = render(<TestReviewsSection reviews={reviews} />);
    expect(queryByText('Тестовый отзыв 2')).toBeTruthy();
    expect(queryByText('Тестовый отзыв 1')).toBeTruthy();
  });

});