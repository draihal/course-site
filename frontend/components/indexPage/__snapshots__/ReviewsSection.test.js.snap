// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsSection render correctly ReviewsSection component 1`] = `
<div
  className="offset"
  id="reviews"
>
  <div
    className="jumbotron"
  >
    <div
      className="container"
    >
      <div
        className="col-12 text-center"
      >
        <h3
          className="heading"
        >
          Отзывы
        </h3>
        <div
          className="heading-underline"
        >
           
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-6 reviews"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4"
            >
              <img
                src="static/default_user.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <blockquote>
                <i
                  className="fas fa-quote-left"
                >
                   
                </i>
                <i>
                  Тестовый отзыв 2
                </i>
                <hr
                  className="reviews-hr"
                />
                <cite>
                  — 
                  Student2
                   
                  Test2
                </cite>
              </blockquote>
            </div>
          </div>
        </div>
        <div
          className="col-md-6 reviews"
        >
          <div
            className="row"
          >
            <div
              className="col-md-4"
            >
              <img
                src="static/default_user.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <blockquote>
                <i
                  className="fas fa-quote-left"
                >
                   
                </i>
                <i>
                  Тестовый отзыв 1
                </i>
                <hr
                  className="reviews-hr"
                />
                <cite>
                  — 
                  Student1
                   
                  Test1
                </cite>
              </blockquote>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-12 narrow text-center"
  >
    <p
      className="lead"
    >
      Хочешь узнать больше мнений?
    </p>
    <a
      className="btn btn-secondary btn-md"
      href="#"
    >
      Другие отзывы
    </a>
  </div>
</div>
`;
